pandas/core/indexes/base.py

```diff
@@ -5289,6 +5289,16 @@ def _get_indexer_non_comparable(
         """
         if method is not None:
             other = unpack_nested_dtype(target)
+            if self._is_multi ^ other._is_multi:
+                kind = other.dtype.type if self._is_multi else self.dtype.type
```

Comment:
do tests hit this? e.g. as you didn't change anything

Reply:
One of the affected cases was not tested; just added a test for that case.